src/App.tsx

```diff
@@ -1,26 +1,428 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  USER_ID,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
 
-const USER_ID = 0;
+type FilterName = 'All' | 'Active' | 'Completed';
+type OptionUpdate = 'all' | 'once';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [loading, setLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<string>('');
+
+  const [waiterLoading, setWaiterLoading] = useState<number | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const [activeFilter, setActiveFilter] = useState<FilterName>('All');
+
+  const showError = (text: string) => {
+    setErrorMsg(text);
+    const timerId = window.setTimeout(() => {
+      window.clearTimeout(timerId);
+      setErrorMsg('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(allTodos => {
+        setTodos(allTodos);
+      })
+      .catch(() => showError('Unable to load todos'))
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, errorMsg]);
+
+  function onFilteredTodos(filterName: FilterName): Todo[] {
```

Comment:
It is important to stick to one way of creating functions
```suggestion
  const onFilteredTodos = (filterName: FilterName): Todo[] => {
```